mail/mailer.go

```diff
@@ -28,16 +49,58 @@ type MailerImpl struct {
 func New(server, port, username, password string) MailerImpl {
 	auth := smtp.PlainAuth("", username, password, server)
 	return MailerImpl{
-		Server: server,
-		Port:   port,
-		Auth:   auth,
-		From:   username,
+		Server:      server,
+		Port:        port,
+		Auth:        auth,
+		From:        username,
+		clk:         clock.Default(),
+		csprgSource: realSource{},
+	}
+}
+
+var maxBigInt = big.NewInt(math.MaxInt64) // ???
+
+func (m *MailerImpl) generateMessage(to []string, subject, body string) ([]byte, error) {
+	mid, err := m.csprgSource.Int(rand.Reader, maxBigInt)
+	if err != nil {
+		return nil, err
+	}
+	now := m.clk.Now().UTC()
+	addrs := []string{}
+	for _, a := range to {
+		addrs = append(addrs, strconv.QuoteToASCII(a))
 	}
+	headers := []string{
+		fmt.Sprintf("To: %s", strings.Join(addrs, ", ")),
+		fmt.Sprintf("From: %s", m.From),
+		fmt.Sprintf("Subject: %s", subject),
+		fmt.Sprintf("Date: %s", now.Format("Mon Jan 2 2006 15:04:05 -0700")),
+		fmt.Sprintf("Message-Id: <%s.%s.%s>", now.Format("20060102T150405"), mid.String(), m.From),
+	}
+	for _, h := range headers[1:] {
+		quoted := strconv.QuoteToASCII(h)
+		h = quoted[1 : len(quoted)-1] // remove forced "" quotes
+	}
+	quotedBody := strconv.QuoteToASCII(body)
```

Comment:
Same comment about `\u`. Also, I think there may be other things you need to escape, like lines beginning with `From`.